imblearn/dask/utils.py

```diff
@@ -0,0 +1,78 @@
+import warnings
+
+import numpy as np
+from sklearn.exceptions import DataConversionWarning
+from sklearn.utils.multiclass import _is_integral_float
+
+
+def is_multilabel(y):
+    if not (y.ndim == 2 and y.shape[1] > 1):
+        return False
+
+    if hasattr(y, "unique"):
+        labels = np.asarray(y.unique())
+    else:
+        labels = np.unique(y).compute()
```

Comment:
I've struggled with this check in dask-ml. Depending on where it's called, it's potentially very expensive (you might be loading a ton of data just to check if it's multi-label, and then loading it again to to the training).

Whenever possible, it's helpful to provide an option to skip this check by having the user specify it when creating the estimator, or in a keyword to `fit` (dunno if that applies here).

Reply:
I thought about it. Do you think that having a context manager outside would make sense:

```python
with set_config(avoid_check=True):
    # some imblearn/scikit-learn/dask code
```

Thought, we might get into trouble with issues related to https://github.com/scikit-learn/scikit-learn/pull/18736

It might just be easier to have an optional class parameter that applies only for dask arrays.